packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1527,6 +1560,10 @@ export class ContainerRuntime
 			snapshotWithContents,
 		} = context;
 
+		this.runtimeOptions = runtimeOptions as Readonly<
+			Required<IContainerRuntimeOptionsInternal>
```

Comment:
why make everything required? seems to just add unsafe casting

Reply:
`runtimeOptions` is also `Readonly<Required<...>>`

Reply:
I guess this cast isn't necessary either.

Reply:
Oh of course not because the extension doesn't do anything right now, besides restate the exact same properties  -_-

But it will be necessary once I remove `flushMode` and `enableGroupedBatching`.  I'm going to put it back now.

Reply:
I think you could backfill right now. This is what you mean right
```typescript
		this.runtimeOptions = {
			flushMode: FlushMode.TurnBased,
			enableGroupedBatching: true,
			...runtimeOptions,
		};
```
?

Reply:
Yep.  TS error because it knows that those things are going to get overwritten by `...runtimeOptions`

Reply:
So, you'll want to update the ctor argument to be
`Required<Omit<IContainerRuntimeOptions, "flushMode" | "enableGroupedBatching">> & IContainerRuntimeOptions`
until those are removed in the breaking change. (The `Readonly` modifier isn't doing anything _currently_.)

Reply:
If I do nothing I'll get red squiggles when removing them from Options and can update this line then.  Are you trying to achieve absolute minimal diff in the breaking PR?  This is feeling like near the point of diminishing returns on refinement

Reply:
I went ahead and did it, to minimize the diff in the break PRs - especially since Mario and I would conflict otherwise.  #23309

Reply:
Minimal diffs in break PR has advantages but it is not the goal. One advantage is avoiding conflicts.
What is more important is to be able to use the new pattern throughout our codebase ahead of the break. That helps confirm that we are doing the right thing and that there is a way to do the right thing. We should be deprecating as well and preventing knew internal uses of those things.